File: 71.simplify-path.py
class Solution:
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        paths = []
        for token in path.split("/"):
            if paths and token == "..":
                paths.pop()
            elif token != ".." and token != "." and token != "":
                paths.append(token)
        return "/" + "/".join(paths)

File: 71.simplify-path.js
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function(path) {
    let paths = [];
    path.split("/").forEach(token => {
        if (token == "..") {
            paths.pop();
        } else if (token != "." && token != "") {
            paths.push(token);
        }
    });
    return "/" + paths.join("/");
};